components/net/indexeddb/idb_thread.rs

```diff
@@ -280,19 +302,17 @@ impl IndexedDBManager {
 
                 match self.databases.entry(idb_description.clone()) {
                     Entry::Vacant(e) => {
-                        let db = IndexedDBEnvironment::new(
-                            HeedEngine::new(
-                                idb_base_dir,
-                                &idb_description.as_path(),
-                                self.thread_pool.clone(),
-                            ),
+                        let db = IndexedDBEnvironment::new(SqliteEngine::new(
+                            idb_base_dir,
+                            &idb_description,
                             version.unwrap_or(0),
-                        );
-                        let _ = sender.send(db.version);
+                            self.thread_pool.clone(),
+                        ));
+                        let _ = sender.send(db.version().unwrap_or(version.unwrap_or(0)));
                         e.insert(db);
                     },
                     Entry::Occupied(db) => {
-                        let _ = sender.send(db.get().version);
+                        let _ = sender.send(db.get().version().unwrap_or(version.unwrap_or(0)));
```

Comment:
We repeat `version.unwrap_or(0)` several times in this function; can we use store that in a variable?